### DISCUSSION TITLE
API to pull temporal/sanctoral feast day?

### DISCUSSION THREAD

**Discussion by IHSPress** [DivinumOfficium/divinum-officium#4170] [Q&A]
Greetings. We are traditional Catholic publishers (ihspress.com); we follow the Divino Afflatu calendar and among other things maintain Cabrol's Holy Week in print. We are getting close to deploying a new website (God willing) after 20+ years and want to work on a feature where on our home page and top header we can display the feast of the temporal and sanctoral cycles. We're prepared to donate some funds and/or pay developers a reasonable amount (as limited by budget) to support development of an API or equivalent functionality that would allow the website to fetch the calendar info from the ordo for each day of the year. 

The goal would be as follows:

- Pull in temporal and sanctoral feats and commemorations.
- Leverage the functionality in the divinumofficium ordo that puts the predominant feast in bold to place that particular text in a "today's feast" box on the website
- Leverage that same functionality to put the subordinate feast (e.g, the saint of the day where subordinate to the temporal feast, or vice versa as appropriate) in smaller print
- Handle scripture or commemoratio as appropriate

I'd be delighted to take suggestions as to 1) display and 2) how easiest to integrate the code/databases/etc into a web deployment (whether by API or running a local instance of the code).

I'll attempt to past into here a snippet of where I'd like to put something like this in the header of our site. Thanks in advance to all for any comments or suggestions.

![image](https://github.com/user-attachments/assets/b93a5ad4-cd29-492e-b8f1-3f7dacb77872)

**Comment by mbab** [comment]
There is probably unused feature in DO which may suit Your needs  – command=kalendar – which return html snippet with required information. eg.
https://www.divinumofficium.com/cgi-bin/horas/Pofficium.pl?date1=11-17-2024&command=kalendar&version=Divino%20Afflatu%20-%201954

**Reply by IHSPress** [reply]
That looks fantastic. THANK YOU.

For our English-speaking readers, if we wanted to put the results of the command into proper English, are there any tools or files resident within the DO project to render the feast information in English? If not we could easily work on translations and possibly have the script fetch from the English-language file via an additional URL parameter - e.g. 

https://www.divinumofficium.com/cgi-bin/horas/Pofficium.pl?date1=11-17-2024&command=kalendar&version=Divino%20Afflatu%20-%201954&lang=en

Thanks again!

**Comment by mbab** [comment]
The last argument should be `lang1=English`, but currently it prints in Latin. Two things should be done:
1. translation of `[Rank]` section in files in English directories Sancti & Tempora.
2. code improvement to take translation of words as _Dominica minor_, month names etc. from file `Psalterium/Common/Translate.txt`.

I will implement Point 2. (as it was already on my long todo list anyway)
and I strongly encourage you to get involved in completing Point 1.

**Reply by APMarcello3** [reply]
Point 1 underway; @DowryOfMary assisting to copy English titles from horas/sancti to missa/sancti.

**Reply by IHSPress** [reply]
Happy to help with point 1. Since we're new to the project (but not liturgy), what's the best way for us to help?

**Reply by IHSPress** [reply]
Looking at the code a little bit (which I am not yet super-familiar with), are we looking at going through the files here:

web/www/horas/English/Sancti/

and running recursive find & replace for all the relevant terms, such as:

find "Duplex II classis"
replace "Double of the Second Class"

Or do we want those translations in the translate.txt file?

Also, I am not finding where the titles are for tempora...

Many apologies if these dumb questions are more trouble than helpful given my unfamiliarity with the structure of the project

**Reply by mbab** [reply]
There is no need to replace anything after two semicolons;
Rank of day is translated in code based on data in Psalterium/Comment.txt (which should be filled as Latin one)

This look Sancti/ directory is already done,
so now in Tempora/  translation must be provided.
